security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -553,9 +557,18 @@ func (sc *SecretCache) rotate(updateRootFlag bool) {
 			// Send the notification to close the stream if token is expired, so that client could re-connect with a new token.
 			if sc.isTokenExpired() {
```

Comment:
The shouldRefresh() and isTokenExpired() is a bit confusing to me. Perhaps we need to consolidate them in the future..